docs/rules/array-element-newline.md

```diff
@@ -19,6 +19,20 @@ Or an object option (Requires line breaks if any of properties is satisfied. Oth
 * `"multiline": <boolean>` requires line breaks if there are line breaks inside elements. If this is false, this condition is disabled.
 * `"minItems": <number>` requires line breaks if the number of elements is at least the given integer. If this is 0, this condition will act the same as the option `"always"`. If this is `null` (the default), this condition is disabled.
 
+Alternatively, different configurations can be specified for array expressions and array patterns:
+
+```json
+{
+    "array-element-newline": ["error", {
+        "ArrayExpression": "consistent",
+        "ArrayPattern": { "minItems": 3 },
+    }]
+}
+```
+
+* `"ArrayExpression"` configuration for array expressions (if unspecified, this rule will not apply to array expressions)
+* `"ArrayPattern"` configuration for array patterns (if unspecified, this rule will not apply to array patterns)
```

Comment:
Currently the docs don't describe anywhere what an array pattern is, and I'm not sure that term would be common knowledge among people who aren't necessarily familiar with AST node types. Could you add some explanation there, and include patterns in a few of the examples?

Reply:
Yep! Just added a bit of clarification to what array patterns are (copied from object-curly-newline which already has separate pattern/expression options) as well as some examples.